validator_client/src/doppelganger_service.rs

```diff
@@ -0,0 +1,159 @@
+use crate::beacon_node_fallback::{BeaconNodeFallback, RequireSynced};
+use crate::initialized_validators::DOPPELGANGER_DETECTION_EPOCHS;
+use crate::validator_store::ValidatorStore;
+use environment::RuntimeContext;
+use slog::{crit, error, info, trace, warn};
+use slot_clock::SlotClock;
+use std::sync::Arc;
+use tokio::time::{interval_at, Duration, Instant};
+use types::{ChainSpec, Epoch, EthSpec};
+
+#[derive(Clone)]
+pub struct DoppelgangerService<T: SlotClock, E: EthSpec> {
+    pub slot_clock: T,
+    pub validator_store: ValidatorStore<T, E>,
+    pub beacon_nodes: Arc<BeaconNodeFallback<T, E>>,
+    pub context: RuntimeContext<E>,
+}
+
+impl<T: 'static + SlotClock, E: EthSpec> DoppelgangerService<T, E> {
+    pub fn start_update_service(self, spec: &ChainSpec) -> Result<(), String> {
+        let log = self.context.log().clone();
+
+        let slot_duration = Duration::from_secs(spec.seconds_per_slot);
+        let duration_to_next_slot = self
+            .slot_clock
+            .duration_to_next_slot()
+            .ok_or("Unable to determine duration to next slot")?;
+
+        info!(
+            log,
+            "Doppelganger detection service started";
+            "next_update_millis" => duration_to_next_slot.as_millis()
```

Comment:
```suggestion
            "next_update" => ?duration_to_next_slot
```

I recently found out that this provides a pretty nice output. Might be a little nicer for users?